### ISSUE TITLE
no-restricted-imports has problems with index imports

### ISSUE THREAD

**Issue by InsOpDe** [eslint/eslint#14990]
**Tell us about your environment**

eslint --env-info
Environment Info:

Node version: v14.17.3
npm version: v7.21.0
Local ESLint version: Not found (its 7.32.0, doesnt resolve properly because of probably pnpm)
Global ESLint version: Not found
Operating System: linux 5.11.0-27-generic

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
* @typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
rules:
  no-restricted-imports:
    - 2
    - paths: []
      patterns:
        - group:
            - "^/src/index"
            - "^src/index"
            - "^/index"
            - "^index"
          message: "No absolute imports"
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```ts
import { EIcon, EIconColor } from "index"
import { MarkdownEditor } from "./index"
```

```bash
eslint src --ext .tsx,.ts,.js
```

**What did you expect to happen?**
It should not mark `import { MarkdownEditor } from "./index"` as error, but mark `import { EIcon, EIconColor } from "index"` as error.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**
```
/home/marcel/Projects/some-project/libraries/react-components/src/components/markdown-editor/stories.tsx
  2:1  error  './index' import is restricted from being used by a pattern. No absolute imports  no-restricted-impors
```

**Steps to reproduce this issue:**

1. Create above ruleset
2. Create above imports
3. Run Eslint

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by mdjermanovic** [comment]
Hi @InsOpDe, thanks for the issue!

> ```
> 2:1  error  './index' import is restricted from being used by a pattern. No absolute imports  no-restricted-impors
> ```

I can't reproduce this locally with your configuration, the code example does not produce any linting errors.

Here's an [online demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IFwibm8tcmVzdHJpY3RlZC1pbXBvcnRzXCI6IFtcbiAgICAyLFxuICAgIHtcbiAgICAgIFwicGF0aHNcIjogW10sXG4gICAgICBcInBhdHRlcm5zXCI6IFtcbiAgICAgICAge1xuICAgICAgICAgIFwiZ3JvdXBcIjogW1xuICAgICAgICAgICAgXCJeL3NyYy9pbmRleFwiLFxuICAgICAgICAgICAgXCJec3JjL2luZGV4XCIsXG4gICAgICAgICAgICBcIl4vaW5kZXhcIixcbiAgICAgICAgICAgIFwiXmluZGV4XCJcbiAgICAgICAgICBdLFxuICAgICAgICAgIFwibWVzc2FnZVwiOiBcIk5vIGFic29sdXRlIGltcG9ydHNcIlxuICAgICAgICB9XG4gICAgICBdXG4gICAgfVxuICBdXG4qL1xuXG5pbXBvcnQgeyBFSWNvbiwgRUljb25Db2xvciB9IGZyb20gXCJpbmRleFwiXG5pbXBvcnQgeyBNYXJrZG93bkVkaXRvciB9IGZyb20gXCIuL2luZGV4XCIiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19) with the same configuration and code, and it's lint-free.

**Comment by mdjermanovic** [comment]
This rule uses [ignore](https://www.npmjs.com/package/ignore) package v4.0.6, and the patterns from the configuration are passed unchanged to `ignore().add()`. I'm not sure if `^` has any particular meaning in patterns.

Would this configuration work for your use case?

```yaml
rules:
  no-restricted-imports:
    - 2
    - paths: []
      patterns:
        - group:
            - "//src/index"
            - "/src/index"
            - "//index"            
            - "/index"
          message: "No absolute imports"
```

An example with the same config in a comment:

```js
/* eslint "no-restricted-imports": [
      2,
      {
        "paths": [],
        "patterns": [
          {
            "group": [
              "//src/index",
              "/src/index",
              "//index",
              "/index"
            ],
            "message": "No absolute imports"
          }
        ]
      }
    ]
*/

import "index"; // error
import "/index"; // error
import "src/index"; // error
import "/src/index"; // error

import "./index"; // ok
import "../index"; // ok
import "./src/index"; // ok
import "../src/index"; // ok

```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IFwibm8tcmVzdHJpY3RlZC1pbXBvcnRzXCI6IFtcbiAgICAgIDIsXG4gICAgICB7XG4gICAgICAgIFwicGF0aHNcIjogW10sXG4gICAgICAgIFwicGF0dGVybnNcIjogW1xuICAgICAgICAgIHtcbiAgICAgICAgICAgIFwiZ3JvdXBcIjogW1xuICAgICAgICAgICAgICBcIi8vc3JjL2luZGV4XCIsXG4gICAgICAgICAgICAgIFwiL3NyYy9pbmRleFwiLFxuICAgICAgICAgICAgICBcIi8vaW5kZXhcIixcbiAgICAgICAgICAgICAgXCIvaW5kZXhcIlxuICAgICAgICAgICAgXSxcbiAgICAgICAgICAgIFwibWVzc2FnZVwiOiBcIk5vIGFic29sdXRlIGltcG9ydHNcIlxuICAgICAgICAgIH1cbiAgICAgICAgXVxuICAgICAgfVxuICAgIF1cbiovXG5cbmltcG9ydCBcImluZGV4XCI7IC8vIGVycm9yXG5pbXBvcnQgXCIvaW5kZXhcIjsgLy8gZXJyb3JcbmltcG9ydCBcInNyYy9pbmRleFwiOyAvLyBlcnJvclxuaW1wb3J0IFwiL3NyYy9pbmRleFwiOyAvLyBlcnJvclxuXG5pbXBvcnQgXCIuL2luZGV4XCI7IC8vIG9rXG5pbXBvcnQgXCIuLi9pbmRleFwiOyAvLyBva1xuaW1wb3J0IFwiLi9zcmMvaW5kZXhcIjsgLy8gb2tcbmltcG9ydCBcIi4uL3NyYy9pbmRleFwiOyAvLyBva1xuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

**Comment by InsOpDe** [comment]
Hi @mdjermanovic thanks for the demo, didnt find that feature.
My problem was that I wanted to target
```typescript
import "index"
```
with the rules
```yaml
rules:
  no-restricted-imports:
    - 2
    - paths: []
      patterns:
        - group:       
            - "index"
            - "^index"
          message: "No absolute imports"
```
which didnt work out. Thank to your example I see now that I had to put the `/` before instead of nothing. It wasnt quite intuitive for me. Thanks for clearing that up.

Can be closed!
